siman/core/structure.py

```diff
@@ -1718,8 +1736,9 @@ def reorder_element_groups(self, order = None, inplace = False):
                     typat.append(t)
                     xcart.append(st.xcart[i])
                     old_numbers.append(i)
-                    if None not in st.magmom:
-                        magmom.append(st.magmom[i])
+                    if st.magmom != []:
```

Comment:
this is very bad, use if len(st.magmom) != 0

Reply:
done